src/Stratis.Features.FederatedPeg/TargetChain/WithdrawalExtractor.cs

```diff
@@ -80,25 +74,46 @@ public IWithdrawal ExtractWithdrawalFromTransaction(Transaction transaction, uin
             if (transaction.IsCoinBase)
                 return null;
 
-            // Withdrawal has a specific structure.
-            if (transaction.Outputs.Count != ExpectedNumberOfOutputsNoChange && transaction.Outputs.Count != ExpectedNumberOfOutputsChange)
-                return null;
-
             if (!this.IsOnlyFromMultisig(transaction))
                 return null;
 
             if (!this.opReturnDataReader.TryGetTransactionId(transaction, out string depositId))
                 return null;
 
-            TxOut targetAddressOutput = transaction.Outputs.SingleOrDefault(this.IsTargetAddressCandidate);
-            if (targetAddressOutput == null)
+            // This is not a withdrawal transaction.
+            if (transaction.Outputs.Count < ExpectedNumberOfOutputsNoChange)
                 return null;
 
+            Money withdrawalAmount = null;
+            string targetAddress = null;
+
+            // Cross Chain transfers either has 2 or 3 outputs.
+            if (transaction.Outputs.Count == ExpectedNumberOfOutputsNoChange || transaction.Outputs.Count == ExpectedNumberOfOutputsChange)
+            {
+                TxOut targetAddressOutput = transaction.Outputs.SingleOrDefault(this.IsTargetAddressCandidate);
+                if (targetAddressOutput == null)
+                    return null;
+
+                withdrawalAmount = targetAddressOutput.Value;
+                targetAddress = targetAddressOutput.ScriptPubKey.GetDestinationAddress(this.network).ToString();
+            }
+
+            // Reward distribution trnsactions will have more than 3 outputs.
+            if (transaction.Outputs.Count > ExpectedNumberOfOutputsChange)
```

Comment:
could be `else if`.

Reply:
This `if` can actually just be removed.

Reply:
Sorry you are right an `else` will work as well.